lib/manager/components/deployment/DeploymentTableRow.js

```diff
@@ -0,0 +1,270 @@
+      return
+    }
+    updateVersionForFeedSource(
+      deployment,
+      feedSource,
+      {id: feedSource.latestVersionId}
+    )
+  }
+
+  _togglePinFeedVersion = () => {
+    const {togglePinFeedVersion, deployment, version} = this.props
+    togglePinFeedVersion(deployment, version)
+  }
+
+  _removeVersion = () => {
+    const {deleteFeedVersion, deployment, version} = this.props
+    deleteFeedVersion(deployment, version)
+  }
+
+  render () {
+    const {
+      feedSource,
+      feedVersionPinned,
+      userCanDecrementVersion,
+      userCanEditDeployment,
+      userCanIncrementVersion,
+      userCanRemoveVersion,
+      version
+    } = this.props
+    const {validationResult: result} = version
+    const hasVersionStyle = {cursor: 'pointer'}
+    const noVersionStyle = {color: 'lightGray'}
+    const expired = versionHasExpired(version)
+    const future = versionHasNotBegun(version)
+    return (
+      <tr>
+        <td>
+          <Link to={`/feed/${feedSource.id}`}>{feedSource.name}</Link>
+          {feedVersionPinned && (
+            <div style={{ fontSize: 12, fontStyle: 'italic', margin: 5 }}>
+              <Glyphicon
+                glyph='star'
+                style={hasVersionStyle}
+                alt='star'
+              />
+              {' '}
+              Version pinned for auto-deployment
+            </div>
+          )}
+        </td>
+        <td className='col-md-1 col-xs-2'>
+          <Link to={`/feed/${feedSource.id}/version/${version.version}`}>
+            <small>Version {version.version}</small>
+          </Link>
+          <ButtonToolbar>
+            <Button
+              bsSize='xsmall'
+              bsStyle='default'
+              style={{color: 'black'}}
+              disabled={!userCanDecrementVersion}
+              onClick={this._decrementVersion}>
+              <Glyphicon
+                glyph='menu-left'
+                style={userCanDecrementVersion ? hasVersionStyle : noVersionStyle}
+                title={version.previousVersionId ? 'Previous version' : 'No previous versions'}
+                alt='Previous version' />
+            </Button>
+            <Button
+              bsSize='xsmall'
+              bsStyle='default'
+              style={{color: 'black'}}
+              disabled={!userCanIncrementVersion}
+              onClick={this._incrementVersion}>
+              <Glyphicon
+                glyph='menu-right'
+                style={userCanIncrementVersion ? hasVersionStyle : noVersionStyle}
+                title={version.nextVersionId ? 'Next version' : 'No newer versions'}
+                alt='Next Version' />
+            </Button>
+            <Button
+              bsSize='xsmall'
+              bsStyle='default'
+              disabled={!userCanEditDeployment}
+              style={{color: 'black'}}
+              onClick={this._togglePinFeedVersion}>
+              <Glyphicon
+                glyph={feedVersionPinned ? 'star-empty' : 'star'}
```

Comment:
Hmm why are we using a star here when we are using a pin to indicate pinned status?